src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -174,6 +171,93 @@ class SidebarLinks extends React.PureComponent {
         this.props.onLinkClick();
     }
 
+    showSettingsPage() {
+        if (this.props.isCreateMenuOpen) {
+            // Prevent opening Settings page when click profile avatar quickly after clicking FAB icon
+            return;
+        }
+
+        Navigation.navigate(ROUTES.SETTINGS);
+    }
+
+    showStatusPage() {
+        if (this.props.isCreateMenuOpen) {
+            // Prevent opening Settings page when click profile avatar quickly after clicking FAB icon
+            return;
+        }
+
+        Navigation.setShouldPopAllStateOnUP();
+        Navigation.navigate(ROUTES.SETTINGS_STATUS);
+    }
+
+    renderAvatarWithIndicator() {
+        return (
+            <PressableWithoutFeedback
+                accessibilityLabel={this.props.translate('sidebarScreen.buttonMySettings')}
+                accessibilityRole={CONST.ACCESSIBILITY_ROLE.BUTTON}
+                onPress={this.showSettingsPage}
+            >
+                <OfflineWithFeedback pendingAction={lodashGet(this.props.currentUserPersonalDetails, 'pendingFields.avatar', null)}>
+                    <AvatarWithIndicator
+                        source={UserUtils.getAvatar(this.props.currentUserPersonalDetails.avatar, this.props.currentUserPersonalDetails.accountID)}
+                        tooltipText={this.props.translate('common.settings')}
+                    />
+                </OfflineWithFeedback>
+            </PressableWithoutFeedback>
+        );
+    }
+
+    /**
+     * Render either a sign-in button or an avatar with optional status.
+     * @returns {React.Component}
+     */
+    renderSignInOrAvatarWithOptionalStatus() {
```

Comment:
The same here - this should be a separate component used in the `render` method

Reply:
👍🏻